utils/utils.tsx

```diff
@@ -866,6 +866,98 @@ export function getCaretXYCoordinate(textArea: HTMLTextAreaElement) {
     };
 }
 
+type CoordParams = {
+    quoteButtonPosition: string;
+    rects: DOMRectList;
+    additionalSpaces: {
+        spaceX: number;
+        spaceY: number;
+        spaceForMultilineSelection: number;
+        startingSelectedElementLeft: number;
+    };
+}
+
+export function getQuoteButtonCoords(coordParams: CoordParams) {
+    const {quoteButtonPosition, rects, additionalSpaces} = coordParams;
+    let positionX = rects[0].x - additionalSpaces.spaceX;
+    let positionY = rects[0].y - additionalSpaces.spaceY;
+
+    if (quoteButtonPosition === 'top') {
+        positionY -= additionalSpaces.startingSelectedElementLeft;
+        positionX -= rects[0].width;
+    }
+
+    // handle multiple line selection on bottom quote button
+    if (quoteButtonPosition === 'bottom' && rects.length > 1) {
+        const numberOfLines = UserAgent.isFirefox() ? rects.length - 1 : Math.ceil((rects.length / 2) - 1);
+        positionY += rects[0].height * numberOfLines;
+        positionX = rects[rects.length - 1].width + additionalSpaces.startingSelectedElementLeft + additionalSpaces.spaceForMultilineSelection;
+    }
+    return {positionX, positionY};
+}
+
+export function findParentPostMessage(node?: HTMLElement | null): HTMLElement | null | undefined {
+    if (!node || node.classList.contains('post-message__text-container')) {
+        return node;
+    }
+    return findParentPostMessage(node.parentElement);
+}
```

Comment:
@michelengelen @AshishDhama I am using a recursive function which will be called in every click in the web app. Do you think this might create a potential performance issues in the future if clicked item is deeply nested?

Reply:
IMO this could potentially be very impactful on the performance. Isn't there a better way to do this? Maybe hook a click listener directly to the post content? That way we would also remove a possible ambiguity on the clicked element

Reply:
> Maybe hook a click listener directly to the post content?

That is a potential solution. In order to put the content to `TextEditor` I need to use redux `GlobalState` right? Is there any existing state which I can use?

Reply:
redux wont really help in this case. One thing you could try is to add an event listener to the post component and then fire a custom event and catch it within the texteditor:

```typescript
document.addEventListener(Constants.QUOTE_TEXT, textQuoteHandler)
```